drivers/shared/executor/executor_linux.go

```diff
@@ -247,7 +247,7 @@ func (l *LibcontainerExecutor) wait() {
 			ps = exitErr.ProcessState
 		} else {
 			l.logger.Error("failed to call wait on user process", "error", err)
-			l.exitState = &ProcessState{Pid: 0, ExitCode: 0, Time: time.Now()}
+			l.exitState = &ProcessState{Pid: 0, ExitCode: -2, Time: time.Now()}
```

Comment:
This is a complicated case and I could use feedback.  When the SIGKILL is used, `proc.Wait()` may return with `waitid: no child processes` - I saw this behavior few times in local development.  Interestingly, I saw the same error reported in 0.8.6 in executor logs when I was investing 0.8.6 behavior.

I considered solving this "properly" by setting executor as the process reaper to collect status, so we could always collect exit code status - but it seemed a bigger rabbithole than I'd be welling at his point - unless I hear opinions.

Reply:
It appears the old executor used `1` for the exit code on unexpected `Wait()` errors: https://github.com/hashicorp/nomad/blob/v0.8.6/client/driver/executor/executor.go#L442

I'd be in favor of maintaining that sentinel unless there's a compelling reason not to.

We should absolutely implement the robust approach, but I think for 0.9.0 maintaining the status quo should be our primary goal.

Reply:
Updated to use 1.